src/lib.rs

```diff
@@ -2,17 +2,28 @@ pub mod plugins;
 
 use axum::Router;
 use eyre::{eyre, Result};
+use hyper::Method;
 use plugins::manager::PluginContainer;
-use tower_http::{catch_panic::CatchPanicLayer, trace::TraceLayer};
+use tower_http::{
+    catch_panic::CatchPanicLayer,
+    cors::{Any, CorsLayer},
+    trace::TraceLayer,
+};
 
 pub fn app() -> Result<(PluginContainer<'static>, Router)> {
     let mut container = PluginContainer::new();
     container.load().map_err(|e| eyre!(e))?;
 
+    let cors = CorsLayer::new()
+        .allow_origin(Any)
+        .allow_methods([Method::GET, Method::POST])
+        .allow_headers(Any);
```

Comment:
Check if `Method::OPTIONS` exists. If so, please add it as well, otherwise we're going to run into issues. It goes hand in hand with `Method::POST`.